packages/components/tabs/src/tabs.js

```diff
@@ -0,0 +1,126 @@
+import css from './tabs.css';
+import { customEventNames } from './utils';
+
+import '@tradeshift/elements.icon';
+import '@tradeshift/elements.typography';
+
+customElementDefineHelper(
+	'ts-tabs',
+	class extends TSElement {
+		constructor() {
+			super();
+			this.tabs = [];
+		}
+
+		static get styles() {
+			return [TSElement.styles, unsafeCSS(css)];
+		}
+
+		static get properties() {
+			return {
+				dir: { type: String, reflect: true }
+			};
+		}
+
+		get direction() {
+			return this.dir ? this.dir : this.bodyDir;
+		}
+
+		tabClickHandler(tab, index) {
+			const tabs = this.shadowRoot.host.querySelectorAll('ts-tab');
+			for (let i = 0; i < tabs.length; ++i) {
+				const curTab = tabs[i];
+				const wasSelected = curTab.getAttribute('selected');
+				if (wasSelected !== null && i !== index) {
+					// reflect the change of selected attribute on the component to hide the content
+					curTab.removeAttribute('selected');
+					// deselect the tab in header
+					delete this.tabs[i].selected;
+				} else if (i === index) {
+					// reflect the change of selected attribute on the component to show the content
+					curTab.setAttribute('selected', 'selected');
+					// select the tab in header
+					this.tabs[i].selected = true;
+				}
+			}
+			this.dispatchCustomEvent(customEventNames.TAB_CLICK, { tab, index });
+			this.requestUpdate();
+		}
+
+		iconTemplate(tab) {
+			if (!tab.icon) {
+				return '';
+			}
+			const type = tab.icon === 'ada' ? 'suggested' : 'default';
+			return html`
+				<ts-icon icon="${tab.icon}" type="${type}" size="medium"></ts-icon>
+			`;
+		}
+
+		badgeTemplate(tab) {
+			if (!tab.counter) {
+				return '';
+			}
+			return html`
+				<em class="badge">${tab.counter}</em>
+			`;
+		}
+
+		tabTemplate(tab, index) {
+			/* eslint-disable lit/no-template-bind */
+			return html`
+				<button ?selected="${tab.selected}" @click="${() => this.tabClickHandler(tab, index)}">
+					${this.iconTemplate(tab)}
+					<ts-typography text="${tab.label}" variant="semi-bold"></ts-typography>
+					${this.badgeTemplate(tab)}
+				</button>
+			`;
+			/* eslint-enable */
+		}
+
+		generateHeaderTabs() {
+			const slottedTabs = this.shadowRoot.host.querySelectorAll('ts-tab');
+			this.tabs = Array.from(slottedTabs).map(tabNode => {
+				return {
+					label: tabNode.getAttribute('label'),
+					icon: tabNode.getAttribute('icon'),
+					counter: tabNode.getAttribute('counter'),
+					selected: tabNode.getAttribute('selected') !== null
+				};
+			});
+			this.requestUpdate();
+		}
+
+		slotChangeHandler(e) {
+			const slottedNodes = e.currentTarget.assignedNodes();
+			const isInvalid = slottedNodes.filter(node => node.nodeType !== Node.TEXT_NODE && node.tagName !== 'TS-TAB')
+				.length;
+			if (isInvalid) {
+				throw new Error('You can only use "ts-tab" in "ts-tabs" component!');
+			}
+			this.generateHeaderTabs();
+		}
+
+		firstUpdated(changedProperties) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove the unused function parameter "changedProperties". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.ts.sv/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)